reconcile/utils/ldap_client.py

```diff
@@ -1,39 +1,39 @@
-from contextlib import contextmanager
-
 from ldap3 import Server, Connection, ALL, SAFE_SYNC
-from reconcile.utils.config import get_config
-
-_base_dn = None
-
-
-@contextmanager
-def init(serverUrl):
-    server = Server(serverUrl, get_info=ALL)
-    client = Connection(server, None, None, client_strategy=SAFE_SYNC)
-    try:
-        client.bind()
-        yield client
-    finally:
-        client.unbind()
 
 
-def init_from_config():
-    global _base_dn
+class LdapClient:
+    """
+    LdapClient that wraps search functionality from ldap3 library
+    and exposes through its own method. The client should be used
+    `with` statement to allow context manager to release connection resource
+    appropriately.
+    """
 
-    config = get_config()
+    def __init__(self, base_dn, connection):
+        self.base_dn = base_dn
+        self.connection = connection
 
-    serverUrl = config["ldap"]["server"]
-    _base_dn = config["ldap"]["base_dn"]
-    return init(serverUrl)
+    def __enter__(self):
+        self.connection.bind()
+        return self
 
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.unbind()
 
-def get_users(uids):
-    global _base_dn
-
-    with init_from_config() as client:
+    def get_users(self, uids: list) -> set:
         user_filter = "".join((f"(uid={u})" for u in uids))
-        _, _, results, _ = client.search(
-            _base_dn, f"(&(objectclass=person)(|{user_filter}))", attributes=["uid"]
+        _, _, results, _ = self.connection.search(
+            self.base_dn, f"(&(objectclass=person)(|{user_filter}))", attributes=["uid"]
         )
-        # pylint: disable=not-an-iterable
         return set(r["attributes"]["uid"][0] for r in results)
+
+    @staticmethod
+    def from_settings(settings):
```

Comment:
Please add type hints, noting that you can add forward references to the class name in quotations:

```python
@staticmethod
def from_settings(settings: <type>) -> "LdapClient":
```

I'd also put together a doc string indicating what settings is expecting.